app/src/main/java/ch/epfl/skysync/viewmodel/FlightsViewModel.kt

```diff
@@ -106,9 +107,16 @@ class FlightsViewModel(
           Log.d("FlightsViewModel", "Admin user loaded")
           _currentFlights.value = repository.flightTable.getAll(onError = { onError(it) })
         } else if (_currentUser.value is Pilot || _currentUser.value is Crew) {
-          _currentFlights.value =
+          val fetchedFlights =
               repository.userTable.retrieveAssignedFlights(
                   repository.flightTable, userId ?: UNSET_ID, onError = { onError(it) })
```

Comment:
It was missed during a previous review, but `userId` should not be nullable (take `userId: String` as attribute), thus there will be no need for `userId ?: UNSET_ID` which wouldn't work anyway.